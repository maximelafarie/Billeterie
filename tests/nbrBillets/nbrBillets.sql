-- ============================================================================
-- Fichier          : nbrBillets.sql
-- Auteur           : Maxime LAFARIE
-- Date             : Janvier 2014
-- Base				: Billeterie
-- Projet           : Billeterie
-- Role             : Nombre de billets vendus par evenement dans l'ordre
--                    decroissant.
-- ============================================================================

SELECT COUNT(idbillet) AS NBRBILLETS, E.titreevt
FROM BILLET B, EVENEMENT E
WHERE B.idevt = E.idevt
GROUP BY titreevt
ORDER BY NBRBILLETS DESC;