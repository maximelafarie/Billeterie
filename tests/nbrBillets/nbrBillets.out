fichier resultat du test : nbrBillets.out
Contenu de la base :
SQL> @ utilitaires/afficherContenuBaseBilleterie.sql
SQL> -- ==========================
SQL> -- fichier : afficherContenuBaseBilleterie.sql
SQL> -- base : BILLETERIE
SQL> -- auteur(s) : Maxime LAFARIE
SQL> -- auteur(s) : Nicolas DOUSSON
SQL> -- date : 07/01/14
SQL> -- role : Afficher le contenu des differentes tables de la base
SQL> -- projet : Billeterie
SQL> -- ==========================
SQL> 
SQL> prompt Contenu de la table client :
Contenu de la table client :
SQL> 
SQL> SELECT * FROM CLIENT;

     IDCLI NOMCLI          PRENOMCLI       TELCLI     MAILCLI                   
---------- --------------- --------------- ---------- --------------------      
ADRCLI                                                                          
---------------                                                                 
         1 MARTIN          Jean            0505050505 martin@gmail.com          
La Rochelle                                                                     
                                                                                
         2 DUPONT          Michel          0605050505 michel@gmail.com          
Niort                                                                           
                                                                                

SQL> 
SQL> prompt Contenu de la table billet :
Contenu de la table billet :
SQL> 
SQL> SELECT * FROM BILLET;

  IDBILLET CODEBILLET           IDCLI    IDTARIF IDSITERESA      IDEVT          
---------- --------------- ---------- ---------- ---------- ----------          
         1 15ED68EP10EPD            1          1          1          1          
         2 85PO36DM19CBD            1          3                     2          
         3 89X736DM55CZM            1          3                     2          
         4 07ED77XD10EFR            2          2          2          3          

SQL> 
SQL> prompt Contenu de la table evenement :
Contenu de la table evenement :
SQL> 
SQL> SELECT * FROM EVENEMENT;

     IDEVT TITREEVT        DATEEVT    TYPEEVT    IDSALLE                        
---------- --------------- ---------- ------- ----------                        
         1 Les Joyaux      01-02-2014 sport            1                        
         2 Clipto          10-04-2014 concert          2                        
         3 Sketch          02-04-2014 comedie          7                        

SQL> 
SQL> prompt Contenu de la table salle :
Contenu de la table salle :
SQL> 
SQL> SELECT * FROM SALLE;

   IDSALLE NOMSALLE        ADRSALLE                                             
---------- --------------- --------------------                                 
         1 CARLINE         La Rochelle                                          
         2 Melrane         Bordeaux                                             
         7 LACCLAMEUR      Niort                                                

SQL> 
SQL> prompt Contenu de la table tarif :
Contenu de la table tarif :
SQL> 
SQL> SELECT * FROM TARIF;

   IDTARIF       PRIX TYPETARI                                                  
---------- ---------- --------                                                  
         1        100 plein                                                     
         3         85 plein                                                     
         2         50 enfant                                                    

SQL> 
SQL> prompt Contenu de la table siteresa :
Contenu de la table siteresa :
SQL> 
SQL> SELECT * FROM SITERESA;

IDSITERESA NOMSITERESA          URLSITERESA                                     
---------- -------------------- --------------------                            
         1 LeMoinsCher          www.lmc.com                                     
         2 eBill                www.ebill.com                                   

SQL> 
SQL> -- ============================================================================
SQL> -- Requete a tester
SQL> -- ============================================================================
SQL> 
SQL> -- Debut de l'affichage du resultat de la requete sur le client SqlPlus
SQL> prompt 'Resultat de la requete'
'Resultat de la requete'
SQL> @ tests/nbrBillets/nbrBillets.sql
SQL> -- ============================================================================
SQL> -- Fichier 	 : nbrBillets.sql
SQL> -- Auteur		 : Maxime LAFARIE
SQL> -- Date		 : Janvier 2014
SQL> -- Base			     : Billeterie
SQL> -- Projet		 : Billeterie
SQL> -- Role		 : Nombre de billets vendus par evenement dans l'ordre
SQL> -- 		   decroissant.
SQL> -- ============================================================================
SQL> 
SQL> SELECT COUNT(idbillet) AS NBRBILLETS, E.titreevt
  2  FROM BILLET B, EVENEMENT E
  3  WHERE B.idevt = E.idevt
  4  GROUP BY titreevt
  5  ORDER BY NBRBILLETS DESC;

NBRBILLETS TITREEVT                                                             
---------- ---------------                                                      
         2 Clipto                                                               
         1 Sketch                                                               
         1 Les Joyaux                                                           

SQL> 
SQL> -- Fin de l'affichage sur le client SqlPlus
SQL> SET echo OFF
